code/modules/mob/living/carbon/update_status.dm

```diff
@@ -8,15 +8,16 @@
 			create_debug_log("died of damage, trigger reason: [reason]")
 			return
 //		if(paralysis || sleeping || getOxyLoss() > low_oxy_ko || (status_flags & FAKEDEATH) || health <= crit_health)
-		if(paralysis || sleeping || (check_death_method() && getOxyLoss() > 50) || (status_flags & FAKEDEATH) || health <= HEALTH_THRESHOLD_CRIT && check_death_method())
-			if(stat == CONSCIOUS)
-				KnockOut()
-				create_debug_log("fell unconscious, trigger reason: [reason]")
+		if(paralysis || sleeping || (check_death_method() && getOxyLoss() > 50) || (status_flags & FAKEDEATH) || health <= HEALTH_THRESHOLD_FULLCRIT && check_death_method())
+			KnockOut()
+			create_debug_log("fell unconscious, trigger reason: [reason]")
 		else
-			if(stat == UNCONSCIOUS)
+			if(health <= HEALTH_THRESHOLD_CRIT && check_death_method())
+				stat = SOFT_CRIT
```

Comment:
I would prefer to move the setting of state to "Soft Crit" into its own function - that way, whatever effects you need to apply to get instant updates (instead of on-life updates) can be easily inserted, and the state-machine can keep its consistent interface.
I'd also like it if you added the debug log message for this state transition as well, to better assess the history of a mob's health.

Reply:
I assume you mean something similar to the WakeUp() proc?